src/bindings/python/src/openvino/frontend/tensorflow/utils.py

```diff
@@ -0,0 +1,204 @@
+        return res
+    raise Error("Could not create example input by provided shape {}".format(input_shapes))
+
+
+def get_concrete_func(tf_function, example_input, input_needs_packing, error_message, use_example_input=True):
+    if input_needs_packing and not isinstance(example_input, (list, tuple)):
+        example_input = [example_input]
+    try:
+        if use_example_input:
+            if not input_needs_packing and isinstance(example_input, (list, tuple)):
+                concrete_func = tf_function.get_concrete_function(*example_input)
+            else:
+                concrete_func = tf_function.get_concrete_function(example_input)
+
+        else:
+            concrete_func = tf_function.get_concrete_function()
+    except Exception as e:
+        raise Exception(error_message.format(e))
+    return concrete_func
+
+
+def trace_tf_model(model, input_shapes, input_types, example_input):
+    import tensorflow as tf
+    if isinstance(model.__call__, tf.types.experimental.GenericFunction):
+        tf_function = model.__call__
+        input_needs_packing = False
+    elif isinstance(model, tf.types.experimental.GenericFunction):
+        tf_function = model
+        input_needs_packing = False
+    else:
+        # Wrap model to tf.Function
+        @tf.function
+        def tf_function(args):
+            return model(*args)
+        input_needs_packing = True
+
+    if example_input is not None:
+        concrete_func = get_concrete_func(tf_function, example_input, input_needs_packing,
+                                          "Could not trace the TF model with the following error: {}")
+    elif input_shapes is not None:
+        inp = create_example_input_by_user_shapes(input_shapes, input_types)
+        concrete_func = get_concrete_func(tf_function, inp, input_needs_packing,
+                                          "Could not trace the TF model with the following error: {}")
+    else:
+        if isinstance(tf_function, tf.types.experimental.GenericFunction) and \
+                tf_function.input_signature is not None:
+            concrete_func = get_concrete_func(tf_function, None, input_needs_packing,
+                                              "Could not trace the TF model with the following error: {}",
+                                              use_example_input=False)
+        else:
+            input_spec = get_input_spec_from_model(model)
+            concrete_func = get_concrete_func(tf_function, input_spec, input_needs_packing,
+                                              "Could not trace the TF model with the following error: {}.\n"
+                                              "Please provide 'example_input'.")
+
+    return concrete_func
+
+
+def type_supported_by_tf_fe(input_model):
+    import tensorflow as tf
+    # Types that require tracing
+    if isinstance(input_model, (tf.keras.layers.Layer, tf.Module, tf.keras.Model, tf.types.experimental.GenericFunction)):
```

Comment:
do we preserve original input tensor names and output names?

Reply:
Original tensor names are preserved only in case if model `__call__` method is `tf.function` and input is not some complex structure like dictionary. Otherwise tensor names are lost after tracing.